tests/test-utils.c

```diff
@@ -32,7 +32,10 @@ ok (const char *format, ...)
 
   printf ("ok %u - ", ++test_number);
   va_start (ap, format);
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wformat-nonliteral"
```

Comment:
Thanks for the PR, but this is not the right way to avoid this warning. The right way would be to add `__attribute__((format(printf, ...)))` to the `ok()` function (and any other function that has printf-like arguments).

Something like this should do it:

```
__attribute__((format(printf, 1, 2))) static void
ok (const char *format, ...)
{
```

Reply:
Right! I was actually just looking at adding this too, but didn't think it was sufficient, will try to drop the diagnostic pragma once I've done this to confirm it's enough

Reply:
It is indeed sufficient, thanks!